prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTask.java

```diff
@@ -0,0 +1,171 @@
+	}
+
+	KubeapiWorkFlowTask(ApiClient apiClient, KubernetesApi api, String beanName) {
+		this.kubeconfig = "";
+		this.apiClient = apiClient;
+		this.api = api;
+		this.setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("api-group").type(WorkParameterType.TEXT).optional(false)
+				.description("API group of resource").build());
+		params.add(WorkParameter.builder().key("api-version").type(WorkParameterType.TEXT).optional(false)
+				.description("API version of resource").build());
+		params.add(WorkParameter.builder().key("kind-plural-name").type(WorkParameterType.TEXT).optional(false)
+				.description("Plural name of the resource kind. E.g. crontabs").build());
+		params.add(WorkParameter.builder().key("operation").type(WorkParameterType.TEXT).optional(false)
+				.description("Operation type/name. Can be one of " + Arrays.toString(OperationType.values())).build());
+		params.add(WorkParameter.builder().key("resource-json").type(WorkParameterType.TEXT).optional(true)
+				.description("The JSON to be used in create and update operations").build());
+		params.add(WorkParameter.builder().key("resource-name").type(WorkParameterType.TEXT).optional(true)
+				.description("Name of resource for get operation").build());
+		params.add(WorkParameter.builder().key("resource-namespace").type(WorkParameterType.TEXT).optional(true)
+				.description("Namespace of resource for get operation").build());
+		params.add(WorkParameter.builder().key("work-ctx-key").type(WorkParameterType.TEXT).optional(true)
+				.description("In get operation the result is stored in WorkContext with the provided key").build());
+		return params;
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskOutput> getWorkFlowTaskOutputs() {
+		return List.of(WorkFlowTaskOutput.OTHER);
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		try {
+			// Init kube client. We do not initialize it in c'tor because it is used as a
+			// Java bean.
+			// Therefore, failure to create client will fail the application.
+			initClient();
+
+			// Get the required parameters
+			String apiGroup = getRequiredParameterValue(workContext, "api-group");
+			String apiVersion = getRequiredParameterValue(workContext, "api-version");
+			String kindPluralName = getRequiredParameterValue(workContext, "kind-plural-name");
+			String operation = getRequiredParameterValue(workContext, "operation");
+
+			OperationType operationType = OperationType.valueOf(operation.toUpperCase());
+
+			switch (operationType) {
+				case UPDATE:
+					update(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case CREATE:
+					create(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+				case GET:
+					get(workContext, apiGroup, apiVersion, kindPluralName);
+					break;
+			}
+		}
+		catch (Exception e) {
```

Comment:
What exception do we expecting here? Would it make sense to be more specific?

Reply:
no. This method is expected not to throw. specifying an exception type is only useful when u have different handling for different types.

Reply:
What I mean is to be more explicit about what we catch here so it is easier to understand.

We handle 3 exceptions:

> ApiException | MissingParameterException | IOException

Reply:
I don't see how is it easier. I personally think it is wrong. Because we want to prevent ANY exception from being thrown regardless of the code inside the try block. So the code starts with 'catch Exception' and never needs to be updated. But I'll change it.

Reply:
fixed